### PR TITLE
[ETD-16] Add ThinPacketizer

### PR DESCRIPTION
**Pull request by psurm**

### CODE REVIEW

File: src/Packetizer/ThinPacketizer.c [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    // Unused in this implementation
+    (void)self;
+    (void)packetType;
+    (void)source;
+    (void)dataOffset;
+
+    assert(destination <= SPACE_PACKET_MAX_APID);
+    assert(packetPointer != NULL);
+    assert(dataOffset == THIN_PACKET_PRIMARY_HEADER_SIZE);
+    assert(dataSize < SPACE_PACKET_MAX_PACKET_DATA_SIZE);
```
Comment (hcorg) [inline src/Packetizer/ThinPacketizer.c:56]: what about `assert(dataSize > 0)`? I see `dataSize - 1` later in the code

File: src/Packetizer/ThinPacketizer.c [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    const size_t receivedDataSize = readPacketDataLength(packetPointer);
+    if(packetSize != THIN_PACKET_PRIMARY_HEADER_SIZE + receivedDataSize) {
+        if(errorCode != NULL) {
+            *errorCode = Packetizer_ErrorCode_IncorrectPacketSize;
+        }
+        return false;
+    }
+
+    // Save the results
+    *destination = (packetPointer[0] << 8u) | packetPointer[1];
```
Comment (hcorg) [inline src/Packetizer/ThinPacketizer.c:102]: readPacketId for symmetry?

File: src/Packetizer/ThinPacketizer.c (hunk 1 above)
Comment (psurm) [reply src/Packetizer/ThinPacketizer.c:56]: I forgot to copy that assert from default Packetizer. 
Fixed.

File: src/Packetizer/ThinPacketizer.c (hunk 2 above)
Comment (psurm) [reply src/Packetizer/ThinPacketizer.c:102]: Actually the asymmetry is propagated from default Packetizer. There is no such function prototype (readPacketId) in common header (PacketizerInternal.h) for some reason. So, should I leave as is, or should I add the common prototype and edit standard Packetizer as well?

Comment (hcorg) [reply src/Packetizer/ThinPacketizer.c:102]: @Lurkerpas - I don't want to impose any impact on existing code, but this seem as some sort of non-consistency. Opinions?

Comment (Lurkerpas) [reply src/Packetizer/ThinPacketizer.c:102]: I agree with @hcorg . Please add the function for symmetry, but only in the ThinPacketizer.

Comment (psurm) [reply src/Packetizer/ThinPacketizer.c]: fixed

**Review by hcorg — APPROVED** [review]

**Pull request merged**